### PR TITLE
Add some help text

### PR DESCRIPTION
**Pull request by mfairchild365**
This also fixes an issue where the validator was not returning any data that would allow us to easily group similar messages.  Message names sometimes include values of attributes and elements found in the code, which in turn would mean that we would have to create help text for each value.  That is unreasonable, so this PR also includes a fix to parse and group similar message text.

### CODE REVIEW

File: src/Metric.php [hunk 1]
```diff
@@ -92,6 +101,38 @@ public function scan($uri, \DOMXPath $xpath, $depth, Page $page, Metrics $contex
...
 
     /**
+     * Get the machine name for a given error message
+     * 
+     * @param $error_message
+     * @return string
+     */
+    public function getMarkNameFromMessage($error_message)
+    {
+        switch ($error_message) {
```
Comment (kabel) [inline src/Metric.php]: This is very hard to follow. It doesn't look like you are switching based on the value of `$error_message`, but rather the result of an expression `switch (true)` ?

Also the ternary `? true : false` is unecessary.

Comment (mfairchild365) [inline src/Metric.php]: You are correct.  Do you think it would be better as a set of if/else statements, or do you think it would be better to change it to `switch(true)`?

Comment (kabel) [inline src/Metric.php]: Personally, I prefer using if/else when you're wanting to do sequential evaluation of expressions, I think it helps to understand the flow of the logic. However, I understand that switches like this tend to be more readable when many cases are added and the logic still flows top-to-bottom.
